### PR TITLE
added _I expect {string} {memoryValidation} at least one of {string}_…

### PR DESCRIPTION
**Pull request by AlexGalichenko**
added _I expect {string} {memoryValidation} at least one of {string}_ step

added _I expect {string} {memoryValidation} all of {string}_ step

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -13,3 +13,12 @@ export async function dataTable2Object(dataTable: DataTable): Promise<{ [key: st
...
     return obj;
 }
+
+/**
+ * Transform key-value data table to array
+ * @param dataTable
+ * @return {any[]}
+ */
+export function dataTable2Array(dataTable: DataTable): Promise<any[]> {
+    return Promise.all(dataTable.raw().map(([value]) => memory.getValue(value)));
```
Comment (KirillBogdanets) [inline src/utils.ts:23]: maybe `allSettled()` will be more useful here?

File: src/validation.ts [hunk 2]
```diff
@@ -105,11 +106,116 @@ Then(
...
         validation(array, membersArray);
     }
 );
 
+function validateAnyOf(AR: any, ERs: any[], validation: (AR: any, ER: any) => void){
+    const errorCollector: Error[] = [];
+    for (const ER of ERs) {
+        try {
+            validation(AR, ER);
+            return;
```
Comment (KirillBogdanets) [inline src/validation.ts:122]: do you need `return` here?

File: src/utils.ts (hunk 1 above)
Comment (AlexGalichenko) [reply src/utils.ts:23]: We should not care about errors getting values from memory, if case of exception it will be caught on step level.

Comment (KirillBogdanets) [reply src/utils.ts:23]: agree

File: src/validation.ts (hunk 2 above)
Comment (AlexGalichenko) [reply src/validation.ts:122]: yep, basically logic is if at least 1 validation passes we stop executing the step and mark it as passed.

Comment (KirillBogdanets) [reply src/validation.ts:122]: got it

**Review by KirillBogdanets — APPROVED** [review]

**Pull request merged**